vulnerabilities/importers/vulnrichment.py

```diff
@@ -0,0 +1,264 @@
+    metrics = cna_data.get("metrics", []) + [data.get("metrics", [])[0] for data in adp_data]
+    vulnrichment_scoring_system = {
+        "cvssV4_0": SCORING_SYSTEMS["cvssv4"],
+        "cvssV3_1": SCORING_SYSTEMS["cvssv3.1"],
+        "cvssV3_0": SCORING_SYSTEMS["cvssv3"],
+        "cvssV2_0": SCORING_SYSTEMS["cvssv2"],
+        "other": {
+            "ssvc": SCORING_SYSTEMS["ssvc"],
+        },  # ignore kev
+    }
+
+    for metric in metrics:
+        for metric_type, metric_value in metric.items():
+            if metric_type not in vulnrichment_scoring_system:
+                continue
+
+            if metric_type == "other":
+                other_types = metric_value.get("type")
+                if other_types == "ssvc":
+                    content = metric_value.get("content", {})
+                    vector_string, decision = ssvc_calculator(content)
+                    scoring_system = vulnrichment_scoring_system[metric_type][other_types]
+                    severity = VulnerabilitySeverity(
+                        system=scoring_system, scoring_elements=vector_string, value=decision
+                    )
+                    severities.append(severity)
+                # ignore kev
+            else:
+                vector_string = metric_value.get("vectorString")
+                base_score = metric_value.get("baseScore")
+                scoring_system = vulnrichment_scoring_system[metric_type]
+                severity = VulnerabilitySeverity(
+                    system=scoring_system, value=base_score, scoring_elements=vector_string
+                )
+                severities.append(severity)
+
+    # Extract references cpes and ignore affected products
+    cpes = set()
+    for affected_product in cna_data.get("affected", []):
+        if type(affected_product) != dict:
+            continue
+        cpes.update(affected_product.get("cpes") or [])  # TODO Add references cpes
+
+    # TODO ADD reference type
+    references = [
+        Reference(url=ref.get("url"), severities=severities)
+        for ref in cna_data.get("references", [])
+    ]
+
+    cpes_ref = [
+        Reference(
+            reference_id=cpe,
+            url=f"https://nvd.nist.gov/vuln/search/results?adv_search=true&isCpeNameSearch=true&query={cpe}",
+        )
+        for cpe in sorted(list(cpes))
+    ]
+    references.extend(cpes_ref)
+
+    weaknesses = set()
+    for problem_type in cna_data.get("problemTypes", []):
+        descriptions = problem_type.get("descriptions", [])
+        for description in descriptions:
+            cwe_id = description.get("cweId")
+            if cwe_id:
+                weaknesses.add(get_cwe_id(cwe_id))
+
+            description_text = description.get("description")
+            if description_text:
+                pattern = r"CWE-(\d+)"
+                match = re.search(pattern, description_text)
+                if match:
+                    weaknesses.add(int(match.group(1)))
```

Comment:
Do we need this? Are there cases when the `cweId` field is empty or not present, but we have a CWE in the `description_text`? And in cases where the CWE is present in `cweId`, should we skip getting the CWE from the description text?

Reply:
@keshav-space I think we need this because sometimes the `cweId` field is empty, even though the CWE is included in the description. However, I'm not sure if we have an example with different CWEs (`cweId`, `description` ) fields, but it seems safe to check both and remove any duplicates.

`vulnrichment-data1.json`:

```
 "problemTypes": [
    {
      "descriptions": [
        {
          "lang": "en",
          "description": "CWE-502 Deserialization of Untrusted Data"
        }
      ]
    }
  ],
```

`vulnrichment-data3.json`:
```
  "problemTypes": [
    {
      "descriptions": [
        {
          "lang": "en",
          "type": "CWE",
          "cweId": "CWE-79",
          "description": "CWE-79: Improper Neutralization of Input During Web Page Generation ('Cross-site Scripting')"
        }
      ]
    }
  ],
```

Reply:
@ziadhany that makes sense.